src/main/java/io/supertokens/webserver/api/useridmapping/UpdateExternalUserIdInfoAPI.java

```diff
@@ -0,0 +1,105 @@
+ *    obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ *    License for the specific language governing permissions and limitations
+ *    under the License.
+ */
+
+package io.supertokens.webserver.api.useridmapping;
+
+import com.google.gson.JsonObject;
+import io.supertokens.Main;
+import io.supertokens.pluginInterface.RECIPE_ID;
+import io.supertokens.pluginInterface.exceptions.StorageQueryException;
+import io.supertokens.useridmapping.UserIdMapping;
+import io.supertokens.useridmapping.UserIdType;
+import io.supertokens.webserver.InputParser;
+import io.supertokens.webserver.WebserverAPI;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.io.Serial;
+
+public class UpdateExternalUserIdInfoAPI extends WebserverAPI {
+    @Serial
+    private static final long serialVersionUID = -6840794570968339459L;
+
+    public UpdateExternalUserIdInfoAPI(Main main) {
+        super(main, RECIPE_ID.USER_ID_MAPPING.toString());
+    }
+
+    @Override
+    public String getPath() {
+        return "/recipe/userid/external-user-id-info";
+    }
+
+    @Override
+    protected void doPut(HttpServletRequest req, HttpServletResponse resp) throws IOException, ServletException {
+        JsonObject input = InputParser.parseJsonObjectOrThrowError(req);
+
+        String userId = InputParser.parseStringOrThrowError(input, "userId", false);
+        // normalize userId
+        userId = userId.trim();
+
+        if (userId.length() == 0) {
+            throw new ServletException(
+                    new WebserverAPI.BadRequestException("Field name 'userId' cannot be an empty String"));
+        }
+
+        String userIdTypeString = InputParser.parseStringOrThrowError(input, "userIdType", true);
+
+        UserIdType userIdType = UserIdType.ANY;
+
+        if (userIdTypeString != null) {
+            // normalize userIdTypeString
+            userIdTypeString = userIdTypeString.trim();
+
+            if (userIdTypeString.equals("SUPERTOKENS")) {
+                userIdType = UserIdType.SUPERTOKENS;
+            } else if (userIdTypeString.equals("EXTERNAL")) {
+                userIdType = UserIdType.EXTERNAL;
+            } else if (!userIdTypeString.equals("ANY")) {
+                throw new ServletException(new WebserverAPI.BadRequestException(
+                        "Field name 'userIdType' should be one of 'SUPERTOKENS', 'EXTERNAL' or 'ANY'"));
+            }
+        }
+
+        String externalUserIdInfo = InputParser.parseStringOrJSONNullOrThrowError(input, "externalUserIdInfo", false);
+
+        if (externalUserIdInfo != null) {
```

Comment:
We do this check cause it's possible that the function returns `null` when the user have given a JSON Null